Libraries/LibWeb/TrustedTypes/TrustedTypePolicy.cpp

```diff
@@ -238,4 +238,78 @@ WebIDL::ExceptionOr<JS::Value> process_value_with_a_default_policy(TrustedTypeNa
     }
 }
 
+// https://www.w3.org/TR/trusted-types/#get-trusted-type-compliant-string-algorithm
+WebIDL::ExceptionOr<Utf16String> get_trusted_type_compliant_string(TrustedTypeName expected_type, JS::Object& global, Variant<GC::Root<TrustedHTML>, GC::Root<TrustedScript>, GC::Root<TrustedScriptURL>, Utf16String> input, InjectionSink sink, String sink_group)
+{
+    auto& vm = global.vm();
+
+    // 1. If input is an instance of expectedType, return stringified input and abort these steps.
+    switch (expected_type) {
+    case TrustedTypeName::TrustedHTML:
+        if (input.has<GC::Root<TrustedHTML>>())
+            return input.get<GC::Root<TrustedHTML>>()->to_string();
+        break;
+    case TrustedTypeName::TrustedScript:
+        if (input.has<GC::Root<TrustedScript>>())
+            return input.get<GC::Root<TrustedScript>>()->to_string();
+        break;
+    case TrustedTypeName::TrustedScriptURL:
+        if (input.has<GC::Root<TrustedScriptURL>>())
+            return input.get<GC::Root<TrustedScriptURL>>()->to_string();
+        break;
+    default:
+        VERIFY_NOT_REACHED();
```

Comment:
Same point as before about not having a `default` case so that the compiler will warn us if/when a new enum value is added. (Also applies below.)